### ISSUE TITLE
vJunos-switch fails to establish connection

### ISSUE THREAD

**Issue by yennym3** [hellt/vrnetlab#147]
Hi,

I'm currently testing the vJunos-switch device, but I'm encountering difficulties establishing a Telnet connection. It appears that the Telnet connection is not being established properly. Additionally, I have ensured that there is a sufficient amount of CPU and RAM resources available.

**Docker logs:**
```
2023-11-07 12:44:30,702: vrnetlab   DEBUG    Creating overlay disk image
'juniper.conf' -> '/var/tmp/tmp.j04QIfeoh5/config/juniper.conf'
Formatting 'config.img', fmt=qcow2 size=1048576 cluster_size=65536 lazy_refcounts=off refcount_bits=16
losetup: config.img: Warning: file does not fit into a 512-byte sector; the end of the file will be ignored.
mkfs.fat: warning - lowercase labels might not work properly with DOS or Windows
mkfs.fat 4.1 (2017-01-24)
/dev/loop8 has 64 heads and 32 sectors per track,
hidden sectors 0x0000;
logical sector size is 512,
using 0xf8 media descriptor, with 384 sectors;
drive number 0x80;
filesystem has 2 12-bit FATs and 4 sectors per cluster.
FAT size is 1 sector, and provides 87 clusters.
There is 1 reserved sector.
Root directory contains 512 slots and uses 32 sectors.
Volume ID is 12ecdf9e, volume label vmm-data   .
Copying file(s) to config disk config.img
./
./config/
./config/juniper.conf
Cleaning up...
removed '/var/tmp/tmp.j04QIfeoh5/config/juniper.conf'
removed directory '/var/tmp/tmp.j04QIfeoh5/config'
removed directory '/var/tmp/tmp.j04QIfeoh5'
removed directory '/var/tmp/tmp.3sPCSTaEBo'
Config disk config.img created
2023-11-07 12:44:30,961: vrnetlab   DEBUG    Starting vrnetlab VJUNOSSWITCH
2023-11-07 12:44:30,961: vrnetlab   DEBUG    VMs: [<__main__.VJUNOSSWITCH_vm object at 0x7f9ff850f430>]
2023-11-07 12:44:30,975: vrnetlab   DEBUG    VM not started; starting!
2023-11-07 12:44:30,975: vrnetlab   INFO     Starting VJUNOSSWITCH_vm
2023-11-07 12:44:30,976: vrnetlab   DEBUG    number of provisioned data plane interfaces is 0
2023-11-07 12:44:30,976: vrnetlab   DEBUG    qemu cmd: qemu-system-x86_64 -enable-kvm -display none -machine pc -monitor tcp:0.0.0.0:4000,server,nowait -m 5120 -serial telnet:0.0.0.0:5000,server,nowait -drive if=ide,file=/vJunos-switch-23.2R1.14-overlay.qcow2 -smp 4,sockets=1,cores=4,threads=1 -cpu IvyBridge,vme=on,ss=on,vmx=on,f16c=on,rdrand=on,hypervisor=on,arat=on,tsc-adjust=on,umip=on,arch-capabilities=on,pdpe1gb=on,skip-l1dfl-vmentry=on,pschange-mc-no=on,bmi1=off,avx2=off,bmi2=off,erms=off,invpcid=off,rdseed=off,adx=off,smap=off,xsaveopt=off,abm=off,svm=off -drive if=none,id=config_disk,file=/config.img,format=raw -device virtio-blk-pci,drive=config_disk -overcommit mem-lock=off -display none -no-user-config -nodefaults -boot strict=on -machine pc-i440fx-focal,usb=off,dump-guest-core=off,accel=kvm -device piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1.0x2 -smbios "type=1,product=VM-VEX" -device pci-bridge,chassis_nr=1,id=pci.1 -device virtio-net-pci,netdev=p00,mac=0C:00:c0:45:1c:00 -netdev user,id=p00,net=10.0.0.0/24,tftp=/tftpboot,hostfwd=tcp::2022-10.0.0.15:22,hostfwd=udp::2161-10.0.0.15:161,hostfwd=tcp::2830-10.0.0.15:830,hostfwd=tcp::2080-10.0.0.15:80,hostfwd=tcp::2443-10.0.0.15:443
```

**Comment by hellt** [comment]
when process stops after qemu is called this usually means you have host-related issues that presents qemu/kvm to work properly

The only thing you can try is to run the image manually with qemu and try to find some qemu logs

**Comment by yennym3** [comment]
> when process stops after qemu is called this usually means you have host-related issues that presents qemu/kvm to work properly
> 
> The only thing you can try is to run the image manually with qemu and try to find some qemu logs

I'm sorry, it was my mistake for not waiting for the approximately 20 min boot time of the device. Now, what I observe is that once the VM has started, the tap interfaces that should be created by the connection mode = "tc" are not being created. 

```    
    ...
    parser = argparse.ArgumentParser(description="")
    parser.add_argument(
        "--trace", action="store_true", help="enable trace level logging"
    )
    parser.add_argument(
        "--hostname", default="vr-vjunosswitch", help="vJunos-switch hostname"
    )
    parser.add_argument("--username", default="vrnetlab", help="Username")
    parser.add_argument("--password", default="VR-netlab9", help="Password")
    parser.add_argument(
        "--connection-mode", default="tc", help="Connection mode to use in the datapath"
    )
    args = parser.parse_args() 
```  
```  
4048: eth1@if4047: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether da:45:70:cc:59:a2 brd ff:ff:ff:ff:ff:ff link-netnsid 0
4050: eth2@if4049: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 4e:60:a6:1e:8d:39 brd ff:ff:ff:ff:ff:ff link-netnsid 1
4052: eth3@if4051: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether ae:25:e8:bc:4a:65 brd ff:ff:ff:ff:ff:ff link-netnsid 0
```

**Comment by hellt** [comment]
can you show the output of `git log -1`?

**Comment by yennym3** [comment]
> can you show the output of `git log -1`?

Yes, of course:

```
~/vrnetlab$ git log -1
commit 57eef5bc091035cd4c16ca8ab192548e04c7a1dd (HEAD -> master, tag: v0.12.1, origin/master, origin/HEAD)
Merge: eb9bd57 9da695a
Author: Roman Dodin <dodin.roman@gmail.com>
Date:   Fri Sep 22 08:23:54 2023 +0200

    Merge pull request #145 from kakkotetsu/fix-144
    
    fix #144 (vJunos-switch fails to create initial config, if startup-config is none)

**Comment by hellt** [comment]
I failed to find anyone recently facing issues with vjunos switch unfortunately
People seem to be running it just fine...

With this I will have to close this until some one else starts to show similar symptoms
